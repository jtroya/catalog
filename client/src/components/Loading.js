import React from 'react';
import PropTypes from 'prop-types';
import { Preloader, Row, Col } from 'react-materialize';

function Loading({ flashing, size }) {
  return (
      <Row className="container">
        <Col s={ 12 } m={ 12 }>
          <div className="center-align">
            <Preloader flashing={ flashing } size={ size } />
          </div>
        </Col>
      </Row>
  );
}

Loading.propTypes = {
  flashing: PropTypes.bool,
  size: PropTypes.string
};

Loading.defaultProps = {
  flashing: false,
  size: 'medium'
};

export default Loading;
