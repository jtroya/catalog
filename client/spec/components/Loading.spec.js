import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Loading from '../../src/components/Loading';

describe('Loading', () => {
  it('should render the Loading component', () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper.find('Preloader')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = shallow(<Loading />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
