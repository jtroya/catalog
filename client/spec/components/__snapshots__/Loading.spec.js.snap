// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches the snapshot 1`] = `
<Row
  className="container"
>
  <Col
    m={12}
    s={12}
  >
    <div
      className="center-align"
    >
      <Preloader
        active={true}
        color="blue"
        flashing={false}
        size="medium"
      />
    </div>
  </Col>
</Row>
`;
